unlock-app/src/components/interface/locks/Settings/forms/CancellationForm.tsx

```diff
@@ -0,0 +1,267 @@
+      })
+    } else {
+      ToastHelper.error('Form is not valid.')
+    }
+  }
+
+  const getFreeTrialLength = async () => {
+    return await web3Service.freeTrialLength({
+      lockAddress,
+      network,
+    })
+  }
+
+  const getRefundPenaltyBasisPoints = async () => {
+    return await web3Service.refundPenaltyBasisPoints({
+      lockAddress,
+      network,
+    })
+  }
+
+  const [
+    { isLoading: isLoadingFreeTrial, data: freeTrialLength },
+    { isLoading: isLoadingPenalty, data: refundPenaltyBasisPoints },
+  ] = useQueries({
+    queries: [
+      {
+        queryFn: async () => getFreeTrialLength(),
+        onSuccess: (value = 0) => {
+          setAllowTrial(value > 0)
+        },
+        onError: () => {
+          ToastHelper.error('Impossible to retrieve freeTrialLength value.')
+        },
+        queryKey: [
+          'getFreeTrialLength',
+          lockAddress,
+          network,
+          updateRefundPenaltyMutation.isSuccess,
+        ],
+      },
+      {
+        queryFn: async () => getRefundPenaltyBasisPoints(),
+        onSuccess: (value = 0) => {
+          setCancelPenalty(value > 0)
+        },
+        onError: () => {
+          ToastHelper.error(
+            'Impossible to retrieve refundPenaltyBasisPoints value.'
+          )
+        },
+        queryKey: [
+          'refundPenaltyBasisPoints',
+          lockAddress,
+          network,
+          updateRefundPenaltyMutation.isSuccess,
+        ],
+      },
+    ],
+  })
+
+  const isLoading = isLoadingPenalty || isLoadingFreeTrial
+
+  const disabledInput = updateRefundPenaltyMutation.isLoading || disabled
+
+  useEffect(() => {
+    setValue('freeTrialLength', allowTrial ? freeTrialLength ?? 0 : 0, {
+      shouldValidate: true,
+    })
+  }, [
+    allowTrial,
+    freeTrialLength,
+    setValue,
+    updateRefundPenaltyMutation.isLoading,
+    updateRefundPenaltyMutation.isSuccess,
+  ])
+
+  useEffect(() => {
+    if (updateRefundPenaltyMutation.isLoading) return
+
+    const refundPenaltyPercentage = (refundPenaltyBasisPoints ?? 0) / 100 // convert basis points to percentage
+
+    setValue(
+      'refundPenaltyPercentage',
+      cancelPenalty ? refundPenaltyPercentage : 0,
+      {
+        shouldValidate: true,
+      }
+    )
+  }, [
+    cancelPenalty,
+    refundPenaltyBasisPoints,
+    setValue,
+    updateRefundPenaltyMutation.isLoading,
+    updateRefundPenaltyMutation.isSuccess,
+  ])
+
+  if (isLoading) return <CancellationFormPlaceholder />
+
+  return (
+    <form
+      className="flex flex-col gap-12"
+      onSubmit={handleSubmit(onUpdateRefundPenalty)}
+    >
+      <div className="flex flex-col gap-6">
+        <SettingCardDetail
+          title="Allow Trial"
+          description="Member can cancel the membership and a refund will be issue. Typically the refund is issued manually by you, and the amount will be lower due of transaction fee"
+        />
```

Comment:
```suggestion
          description="If you enable a free trial period, users will be able to get a full refund when they cancel their memberships. It is strongly advised to add transfer penalties when enabling free trials to avoid risks of people stealing funds from your contract."
        />
```